tools/serverpod_cli/lib/src/database/migration.dart

```diff
@@ -85,6 +86,30 @@ DatabaseMigration generateDatabaseMigration({
     }
   }
 
+  var tablesToBeDeleted = actions
```

Comment:
This can be moved up, as we collect the tables to be deleted early. So we might just be able to pass that into the "alter table" computation, and then avoid the "double drop".